tools/testing/test_selections.py

```diff
@@ -196,21 +188,39 @@ def calculate_shards(
 ) -> List[Tuple[float, List[ShardedTest]]]:
     must_serial = must_serial or (lambda x: True)
     test_class_times = test_class_times or {}
+
+    # Divide tests into pytest shards
+    if sort_by_time:
+        known_tests = [
+            x
+            for x in tests
+            if get_duration(x, test_file_times, test_class_times) is not None
```

Comment:
Why do we need separate `get_time` and `get_duration` functions?

Reply:
`get_duration` actually calculates the time and is used to populate `get_time`, which just returns the value inside the object.  `get_duration` also returns an optional float while `get_time` returns just a float